syft/workers/base.py

```diff
@@ -1005,7 +1005,8 @@ def serializer(self, workers=None):
             workers.append(self)
         else:
             # self is referenced in self.__known_workers
-            workers = [w for _, w in self._known_workers.items()]
+            # NOTE: for some reason self._known_workers may contain a Plan
```

Comment:
Well caught! Indeed this happens. Maybe BaseWorker -> AbstractWorker to be more generic?